### ISSUE TITLE
How do you create the swing.js file?

### ISSUE THREAD

**Issue by ihh** [gajus/swing#95]
Sorry for what must be a very n00bish question, but how might I create an up-to-date version of the swing.js file that is used in the demo?

From inspecting the code it looks like it was built with webpack, but simply running webpack on the .js files in the src subdirectory doesn't seem to create the same file (it's only about 1/10 the size, and missing the gajus.* namespace, and doesn't work as a drop-in replacement)

I assume this is some trivial npm or webpack command that is so universally known that no-one even documents it... any chance someone could enlighten me?

Cheers!

**Comment by ihh** [comment]
Answering my own question (fumbling towards learning how to use webpack)

The `examples` directory includes a `README.md` that describes how to build the `swing.js` file used by the online examples (or, more precisely, to use webpack to serve up those examples from a locally hosted server). However, this uses the `npm` package for swing, which is rather out-of-date (I wanted to use the `allowMovement` code from #94, and although I could copy across that PR manually, I felt that this was a suboptimal solution)

Instead, after inspecting the `index.js` files in the `examples` and `src` directories, I created a file called `webpack_index.js` in the `src` directory, containing the following line:

    window.swing = require("./index.js");

In order to remove some ES6 syntax (not yet supported by Safari, and throws errors in webpack/uglify) I did a quick search and replace for ES6 arrow function definitions, default arguments, and `let` statements:

    perl -i.bak -pe 's/(\([^\(\)*]*\)) *=>/function $1/g;s/\blet\b/var/g;s/(function.*?)([a-z]+)\b *(= *[^\(\),]+)(.*)/$1$2$4\n  if (typeof($2) === "undefined") { $2 $3 }/g' Card.js Direction.js Stack.js utilities.js

I also manually edited out some ES6 object initalizers of the form `x = { a, b, c }` replacing them with `x = { a: a, b: b, c: c }`. It's a pity to remove ES6 like this but again it doesn't seem to be widely supported yet.

Then, still in `src`, I ran webpack as follows

    webpack --entry ./webpack_index.js swing.js
    webpack --optimize-minimize --entry ./webpack_index.js swing.min.js

This may all be obvious to anyone with even a modicum of proper JS/webpack experience but it was not totally trivial for me, so I'm posting it here in case it's useful to others.
